react-components/src/hooks/types.ts

```diff
@@ -58,3 +64,8 @@ export type Image360AnnotationMappedAssetData = {
   asset: Asset;
   annotationIds: number[];
 };
+
+export type ModelWithRevision = {
+  model: Model3D | undefined;
+  revision: Revision3D | undefined;
+};
```

Comment:
The `ModelWithRevision` type allows both `model` and `revision` properties to be `undefined`. However,  the usage of, it seems that the `model` and `revision` properties should always have defined values when the `ModelWithRevision` object itself is defined. So, it would be more appropriate to have `model` and `revision` are not `undefined`. The `ModelWithRevision` object itself can still be `undefined`.

To reflect this, the type definition should be updated as follows:

```typescript
export type ModelWithRevision = {
  model: Model3D;
  revision: Revision3D;
};
```

This change ensures that when a `ModelWithRevision` object is defined, both `model` and `revision` properties are guaranteed to be defined as well. If there is a need to represent the absence of a model or revision, the entire `ModelWithRevision` object should be `undefined` instead.

Reply:
nice catch.
https://github.com/cognitedata/reveal/pull/4746/commits/f51740e72efab7b49df11e4dc584ae056644e8f9